Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino

```diff
@@ -1,15 +1,63 @@
 // TODO:
 // Serial Interface - Needs to be initialized, look into plotting data with the data streamer add-on in Excel.
-// Temeperature Sensor - Needs to be initialized, periodically poll sensor, store temperature in variable, print out to serial monitor, look into kalman filters for reducing sensor noise.
+// Rida -In Progress- Temperature Sensor - Needs to be initialized, periodically poll sensor, store temperature in a variable, print out to serial monitor, look into Kalman filters for reducing sensor noise.
 // IMU - Needs to be initialized, periodically poll sensor, store Z-axis angle in variable, take initial value as 0 deg print out to serial monitor, look into kalman filters for reducing sensor noise.
 // Stir Bar Motor - Needs to be initialized, just constantly turn at 80% speed for now, we'll tune speed later.
 // Servo Motor - Needs to be initialized, turn 180 deg clockwise once at start to dump reactants, wait 1 sec to finish dumping, turn back 180 deg counter-clockwise to return to upright position, set speed to 100% for now, tune later.
 // MicroSD Card - Store all variable data to MicroSD Card on microcontroller, allow for each run to be saved under a different filename (potentially in CSV format) so we can keep track of everything.
 
+#include <OneWire.h>
+#include <DallasTemperature.h>
+#define ONE_WIRE_BUS 23 // pin for the DS18B20 data line
+OneWire oneWire(ONE_WIRE_BUS); // create a OneWire instance to communicate with the senso
+DallasTemperature sensors(&oneWire); // pass oneWire reference to Dallas Temperature sensor
+
+// variable to store temperature
+float temperatureC;
+// KALMAN FILTER variables
+float x_k; // State estimate: current best estimate of true temp
+float p_k; // Error covariance: reps the uncertainity in the state estimate
+// Process noise and measurement noise
+float q; // Process noise covariance
+float r; // Measurement noise covariance
+
 void setup() {
+  Serial.begin(9600);  // start serial communication (adjust baud rate as needed)
+  sensors.begin();   // initialize the DS18B20 sensor
 
+  // Initialize Kalman filter parameters
+  x_k = 0.0; // Initial state estimate
+  p_k = 1.0; // Initial error covariance
+  q = 0.01;  // Process noise covariance 
+  r = 0.1;   // Measurement noise covariance 
```

Comment:
Will any of the initial parameters you set need to be tuned as we do trial runs?

Reply:
Yes, these parameters need to be tuned based on the characteristics of the system and the noise in our measurements. All of these would be determined while doing trial run. like for example q needs to be increased in the case of our system being very dynamic.